algorithms/src/std_algorithms/impl/Kokkos_Constraints.hpp

```diff
@@ -215,6 +230,40 @@ KOKKOS_INLINE_FUNCTION void expect_valid_range(IteratorType first,
   (void)last;
 }
 
+//
+// Check if kokkos iterators are overlapping
+//
+template <typename IteratorType1, typename IteratorType2>
+KOKKOS_INLINE_FUNCTION void expect_no_overlap(
+    [[maybe_unused]] IteratorType1 first, [[maybe_unused]] IteratorType1 last,
+    [[maybe_unused]] IteratorType2 s_first,
+    [[maybe_unused]] IteratorType2 s_last) {
```

Comment:
Do you mean to take `s_last` as argument rather than computing it with something like `advance(s_first, distance(first, last))`?

Reply:
That is true. We no longer need `s_last`. I will fix that.